config.libsonnet

```diff
@@ -4,14 +4,17 @@ local slo = import 'slo-libsonnet/slo.libsonnet';
   _config+:: {
     SLOs: {
       apiserver: {
-        // This is templating a Multiple Burn Rate Alerts for Kubernetes Apiservers.
-        // We will alert on burning too much error budget over 30 days.
-        // By default we have 99% availability (1% unavailability = 7h12m in 30d).
-        errors: slo.errorburn({
-          metric: 'apiserver_request_total',
-          selectors: [$._config.kubeApiserverSelector],
-          errorBudget: 1 - 0.99,
-        }),
+        days: 7,  // The number of days we alert on burning too much error budget for.
+        target: 0.99, // The target percentage of availability between 0-1. (0.99 = 99%, 0.999 = 99.9%)
+
+        // Only change these windows when you really understand multi burn rate errors.
+        // Otherwise change days and target above to make high-level changes.
+        windows: [
+          { severity: 'critical', 'for': '2m', long: '1h', short: '5m', factor: 2 * (24 * $._config.SLOs.apiserver.days) / 100 },
+          { severity: 'critical', 'for': '15m', long: '6h', short: '30m', factor: 1 * (24 * $._config.SLOs.apiserver.days) / 100 },
+          { severity: 'warning', 'for': '1h', long: '1d', short: '2h', factor: 0.5 * (24 * $._config.SLOs.apiserver.days) / 100 },
+          { severity: 'warning', 'for': '3h', long: '3d', short: '6h', factor: 0.1 * (24 * $._config.SLOs.apiserver.days) / 100 },
```

Comment:
I don't quite understand the logic behind the factor calculation. With the given defaults, the factor is 0.84 for a the 1d long window and 0.168 for the 3d long window. That means we would alert for the longer windows even if we burn our error budget at a lower rate than our error budget allows. What am I missing?

Reply:
This was my attempt in making the factors somewhat adjustable to different windows (like 7d or 30d). 
What exactly do you think could we improve here? Or would you say just hardcore these things and have people change it on their own? 
The different values depend on each other, no? Like the factor is tied to how many days its for (2 percent * 24 hours * 30 days / 100 = 14.4) what we use all over the place.

Reply:
For the exact reasoning, I guess we have to go back to the source in chapter 5 of the SRE workbook. IIRC, those values are a bit matter of taste. You want some good spread of reaction times. But saying that you want to know that you have burned more than 10% of your monthly error budget in less than 3d, that's a bit because you like it that way.

The most plausible first order approximation to a different error budget time frame (e.g. weekly instead of monthly) would IMHO be to scale down all durations accordingly, but leave the factor the same, i.e. divide the long window duration by four (or multiply by 7/30, to be precise) and the short window duration and the `FOR` duration.

If the window durations get too short due to that, you might want to tweak things in another way. Perhaps the 10%/5%/2% sequence is pretty good as a started. Then the longest window would always have a factor of 1 and a duration of 1/10th of the SLO "billing period". Then you have to pick subsequently shorter windows for the other alert brackets, paying attention that the factor is never smaller than 1 but also never so big that it doesn't really make sense anymore (like resulting in 100% error rate still being OK).

It's not easy… (o:

Reply:
I've simplified the windows to be for 30d for now. 
I'll create another PR after we're done with this one, to start a discussion about calculating the burnrate windows again.